syntax = "proto3";

message ProtoFile {
    bytes data = 1;
    string filename = 2;
}

message TextField {
    string title = 1;
    string text = 2;
    bool inline = 3;
}

message EmbedContent {
    string title = 1;
    string description = 2;
    string author = 3;
    int32 color = 4;
    ProtoFile snapshot = 5;
    repeated TextField textfield = 6;
}

message Presence {
    string presence = 1;
}

message Request {
    string command = 1;
    int64 user = 2;
    ProtoFile file = 3;
}

message Response {
    EmbedContent embed = 1;
    Presence presence = 2;
    ProtoFile file = 3;
}
